haralyzer/assets.py

```diff
@@ -404,13 +443,24 @@ def time_to_first_byte(self):
         """
         Time to first byte of the page request in ms
         """
-        initial_entry = self.entries[0]
         ttfb = 0
-        for k, v in iteritems(initial_entry['timings']):
-            if k != 'receive':
-                ttfb += v
+        for x in self.entries:
+            if x['response']['status'] == 200:
+                for k, v in iteritems(x['timings']):
+                    if k != 'receive':
+                        ttfb += v
+                break
+            else:
+                ttfb += x['time']
+
         return ttfb
 
+        #ttfb = 0
```

Comment:
Can we go ahead and just remove this code block instead of commenting?

Reply:
Hi, this is my first commit to open-source project :)

I left the commented part, because maybe is better to leave two functions for TTFB.

I'm not sure, if TTFB should include the redirect times or not.

My change, deals with redirects - sum the time of all redirected entries before first with code 200.

I've got negative timings for TTFB with the old version, because of "-1" values. Maybe if we just check for positive number before sum, this will be fixed.

If you like, I can remove the commented part, or maybe will be better to have two functions, one summing redirect times, one not.

Reply:
Congrats on your first contribution to open source!

Reading through the definitions of TTFB, it is the time that the first byte of the actual PAGE is received by the client. As such, my original logic of only looking at the first entry was actually incorrect, as it could have been a redirect. Your logic is correct (i.e. - all redirects should count as part of TTFB until we get a 200 and the first byte of the body). Go ahead and just remove the commented code, as it is no longer useful. In a worst case scenario (for instance, if we realize we are totally wrong and have broken everything) we have all of the changes in our git history and can revert as necessary.